docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,224 @@
+```
+## Connectors
+The Fybrik website [describes](https://fybrik.io/v1.1/concepts/connectors/) what are connectors and their types.
+There are three different connectors types:
+- [Data Catalog](https://github.com/fybrik/fybrik/blob/master/connectors/api/datacatalog.spec.yaml) connector
+- [Policy Manager](https://github.com/fybrik/fybrik/blob/master/connectors/api/policymanager.spec.yaml) connector
+- [Credential Management](https://github.com/fybrik/fybrik/blob/master/pkg/vault/vault_interface.go) connector 
+(TODO add OpenAPI definition, see the issue [#1761](https://github.com/fybrik/fybrik/issues/1761) )
+
+Fybrik manager communicates with connectors over REST APIs. 
+
+REST doesn’t provide any specific versioning guidelines, however the more commonly used approaches fall into three 
+categories:
+- URI Versioning
+- Versioning using Custom Request Header
+- Versioning using the “Accept” header
+For examples and explanations, see [REST API Versioning](https://restfulapi.net/versioning/)
+
+For connectors REST API, we suggest using URI versioning as the most commonly used and easiest to check.
+Which means, all REST requests to for example, version 3 will have URI prefix `/v3`. Currently, Fybrik connectors do not
+have the versioning prefix. We can assume that requests without the prefix relate to the v1 version. 
+
+*Notes:* 
+- The current implementation of controllers REST API does not support any versioning. The new incompatible versions will 
+start from the "/v2" prefix, and we will assume that missing version information is the current (old) version.
+- We do not expect significant API changes. Most of the possible changes are covered by the Taxonomy
+- OpenApi [specification](https://swagger.io/specification/) includes `version` entry in the `Info`, however, the object 
+> "provides metadata about the API. The metadata MAY be used by the clients if needed, and MAY be presented in editing or 
+documentation generation tools for convenience." 
+
+We will update the OpenAPI version values, but they cannot be used for API communications.
+
+## Default Taxonomy
+Fybrik widely uses [Taxonomy](https://fybrik.io/v1.1/concepts/taxonomy/) to defines the terms and related values that 
+need to be commonly understood and supported across the components in the system. Taxonomy provides a mechanism for all 
+Fybrik components to interact using a common dialect. Taxonomy defines a set of immutable structural JSON schemas, or 
+"taxonomies" for resources deployed in Fybrik. However, since the taxonomy is meant to be configurable, a taxonomy.json 
+file is referenced from these schemas for any definition that is customizable. The taxonomy.json file is generated from 
+a base taxonomy and zero or more taxonomy layers:
+- The base taxonomy is maintained by the project and includes all the structural definitions that are subject to 
```

Comment:
So this is a key integration point that shouldn't change frequently.  Should the base taxonomy be versioned?

Reply:
yes, and see a suggestion below